src/hotspot/share/prims/jvmtiEnvBase.cpp

```diff
@@ -756,6 +756,17 @@ JvmtiEnvBase::get_thread_state(oop thread_oop, JavaThread* jt) {
   return state;
 }
 
+jint
+JvmtiEnvBase::get_thread_state(oop thread_oop, JavaThread* jt) {
+  jint state = get_thread_state_base(thread_oop, jt);
+
+  if (is_thread_carrying_vthread(jt, thread_oop)) {
+    state &= ~JVMTI_THREAD_STATE_RUNNABLE;
+    state |= JVMTI_THREAD_STATE_WAITING | JVMTI_THREAD_STATE_WAITING_INDEFINITELY;
```

Comment:
How about a comment here:

"Clear RUNNABLE state and add WAITING state because..."

Reply:
Thanks. Added comment.